(cli-node, repo-tools): refactor tests to avoid mock-fs (#20505)
backstage/backstage: .changeset/violet-houses-stare.md

---

> '@backstage/repo-tools': minor
> '@backstage/cli-node': minor
> ---
> 
> Replace mock-fs in test from `cli-node`, `repo-tools` packages

Reviewer: ```suggestion
'@backstage/repo-tools': patch
'@backstage/cli-node': patch
---

Replace `mock-fs` with internal mock utility in tests
```
